### PR TITLE
Update to RAPIDS 24.10

### PR DESCRIPTION
**Pull request by cwharris**
## Description

Closes https://github.com/nv-morpheus/MRC/issues/478
Requires https://github.com/nv-morpheus/utilities/pull/75

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/nv-morpheus/Morpheus/blob/main/docs/source/developer_guide/contributing.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: cmake/dependencies.cmake [hunk 1]
```diff
@@ -48,18 +48,8 @@ morpheus_utils_configure_cccl()
 # =================
 morpheus_utils_configure_rmm()
 
-# gflags
-# ======
-rapids_find_package(gflags REQUIRED
-  GLOBAL_TARGETS gflags
-  BUILD_EXPORT_SET ${PROJECT_NAME}-exports
-  INSTALL_EXPORT_SET ${PROJECT_NAME}-exports
-)
```
Comment (dagardner-nv) [inline cmake/dependencies.cmake:57]: Why are we removing gflags, aren't we using it?

File: cpp/mrc/src/internal/utils/contains.hpp [hunk 2]
```diff
@@ -29,13 +29,15 @@ bool contains(const ContainerT& container, const KeyT& key)
...
-                                         typename C::reference>
+class KeyIterator
 {
   public:
+    using iterator_category = std::bidirectional_iterator_tag;
+    using value_type        = C::key_type;
+    using difference_type   = C::difference_type;
+    using pointer           = C::pointer;
+    using reference         = C::reference;
+
```
Comment (dagardner-nv) [inline cpp/mrc/src/internal/utils/contains.hpp:40]: Is this change unrelated to the upgrade?

File: cmake/dependencies.cmake (hunk 1 above)
Comment (cwharris) [reply cmake/dependencies.cmake:57]: I moved the find to `morpheus_utils_configure_glog`.

File: cpp/mrc/src/internal/utils/contains.hpp (hunk 2 above)
Comment (cwharris) [reply cpp/mrc/src/internal/utils/contains.hpp]: This wouldn't compile without the change. I think it's related to the gxx version update.

**Review by AnuradhaKaruppiah — APPROVED** [review]

File: ci/conda/recipes/libmrc/conda_build_config.yaml [hunk 3]
```diff
@@ -30,4 +30,4 @@ python:
 
 # Setup the dependencies to build with multiple versions of RAPIDS
 rapids_version: # Keep around compatibility with current version -2
-  - 24.02
+  - 24.06
```
Comment (dagardner-nv) [inline ci/conda/recipes/libmrc/conda_build_config.yaml:33]: Should this be `24.10`?

**Review by dagardner-nv — APPROVED** [review]
Just one question

File: ci/conda/recipes/libmrc/conda_build_config.yaml (hunk 3 above)
Comment (cwharris) [reply ci/conda/recipes/libmrc/conda_build_config.yaml]: thanks!

**Pull request merged**